netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -73,6 +73,21 @@
  * A Netty-based {@link ClientTransport} implementation.
  */
 class NettyClientTransport implements ClientTransport {
+
+  private final ConcurrentHashMap<String, Http2Headers> methodToDefaultHeaders =
```

Comment:
In theory string should be better if we were creating many instances of MethodDescriptor. I don't think theres a meaningful performance difference